tests/saitests/py3/sai_qos_tests.py

```diff
@@ -789,40 +789,45 @@ def runTest(self):
 
             # There are packet leak even port tx is disabled (18 packets leak on TD3 found)
             # Hence we send some packet to fill the leak before testing
-            for dscp, _ in dscp_to_pg_map.items():
-                pkt = self._build_testing_pkt(
-                    active_tor_mac=active_tor_mac,
-                    standby_tor_mac=standby_tor_mac,
-                    active_tor_ip=active_tor_ip,
-                    standby_tor_ip=standby_tor_ip,
-                    inner_dscp=dscp,
-                    outer_dscp=0,
-                    dst_ip=dst_port_ip
-                )
-                send_packet(self, src_port_id, pkt, 20)
-            time.sleep(10)
-
-            for dscp, pg in dscp_to_pg_map.items():
+            if asic_type != 'mellanox':
+                for dscp, _ in dscp_to_pg_map.items():
+                    pkt = self._build_testing_pkt(
+                        active_tor_mac=active_tor_mac,
+                        standby_tor_mac=standby_tor_mac,
+                        active_tor_ip=active_tor_ip,
+                        standby_tor_ip=standby_tor_ip,
+                        inner_dscp=dscp,
+                        outer_dscp=0,
+                        dst_ip=dst_port_ip
+                    )
+                    send_packet(self, src_port_id, pkt, 20)
+                time.sleep(10)
```

Comment:
avoid the unreliable sleep and use retry instead

Reply:
The sleep is from the original code and is for the packets leak on TD3, I suggest we keep it here.

Reply:
Yes, the sleep is for packet leak. Please keep the sleep. Thanks!